File: plugin-azure-server-arm/src/main/java/jetbrains/buildServer/clouds/azure/arm/AzureConstants.java
/*
 * Copyright 2000-2016 JetBrains s.r.o.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jetbrains.buildServer.clouds.azure.arm;

/**
 * ARM constants.
 */
public class AzureConstants {

    public static final String TENANT_ID = "tenantId";
    public static final String CLIENT_ID = "clientId";
    public static final String CLIENT_SECRET = "clientSecret";
    public static final String SUBSCRIPTION_ID = "subscriptionId";
    public static final String LOCATION = "location";
    public static final String IMAGE_URL = "imageUrl";
    public static final String OS_TYPE = "osType";
    public static final String NETWORK_ID = "networkId";
    public static final String SUBNET_ID = "subnetId";
    public static final String MAX_INSTANCES_COUNT = "maxInstances";
    public static final String VM_SIZE = "vmSize";
    public static final String VM_NAME_PREFIX = "vmNamePrefix";
    public static final String VM_PUBLIC_IP = "vmPublicIp";
    public static final String VM_USERNAME = "vmUsername";
    public static final String VM_PASSWORD = "vmPassword";
    public static final String TAG_SERVER = "teamcity-server";
    public static final String TAG_PROFILE = "teamcity-profile";
    public static final String TAG_SOURCE = "teamcity-source";

    public String getTenantId() {
        return TENANT_ID;
    }

    public String getClientId() {
        return CLIENT_ID;
    }

    public String getClientSecret() {
        return CLIENT_SECRET;
    }

    public String getSubscriptionId() {
        return SUBSCRIPTION_ID;
    }

    public String getLocation() {
        return LOCATION;
    }

    public String getImageUrl() {
        return IMAGE_URL;
    }

    public String getOsType() {
        return OS_TYPE;
    }

    public String getMaxInstancesCount() {
        return MAX_INSTANCES_COUNT;
    }

    public String getNetworkId() {
        return NETWORK_ID;
    }

    public String getSubnetId() {
        return SUBNET_ID;
    }

    public String getVmSize() {
        return VM_SIZE;
    }

    public String getVmNamePrefix() {
        return VM_NAME_PREFIX;
    }

    public String getVmPublicIp() {
        return VM_PUBLIC_IP;
    }

    public String getVmUsername() {
        return VM_USERNAME;
    }

    public String getVmPassword() {
        return VM_PASSWORD;
    }
}

File: plugin-azure-server-arm/src/main/kotlin/jetbrains/buildServer/clouds/azure/arm/AzureConstants.kt
/*
 * Copyright 2000-2017 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jetbrains.buildServer.clouds.azure.arm

import jetbrains.buildServer.clouds.CloudImageParameters

/**
 * ARM constants.
 */
class AzureConstants {
    val environment: String
        get() = ENVIRONMENT

    val tenantId: String
        get() = TENANT_ID

    val clientId: String
        get() = CLIENT_ID

    val clientSecret: String
        get() = CLIENT_SECRET

    val subscriptionId: String
        get() = SUBSCRIPTION_ID

    val deployTarget: String
        get() = DEPLOY_TARGET

    val region: String
        get() = REGION

    val groupId: String
        get() = GROUP_ID

    val imageType: String
        get() = IMAGE_TYPE

    val imageUrl: String
        get() = IMAGE_URL

    val imageId: String
        get() = IMAGE_ID

    val instanceId: String
        get() = INSTANCE_ID

    val osType: String
        get() = OS_TYPE

    val maxInstancesCount: String
        get() = MAX_INSTANCES_COUNT

    val networkId: String
        get() = NETWORK_ID

    val subnetId: String
        get() = SUBNET_ID

    val vmSize: String
        get() = VM_SIZE

    val vmNamePrefix: String
        get() = VM_NAME_PREFIX

    val vmPublicIp: String
        get() = VM_PUBLIC_IP

    val vmUsername: String
        get() = VM_USERNAME

    val vmPassword: String
        get() = VM_PASSWORD

    val reuseVm: String
        get() = REUSE_VM

    val storageAccountType: String
        get() = STORAGE_ACCOUNT_TYPE

    val template: String
        get() = TEMPLATE

    val imagesData: String
        get() = CloudImageParameters.SOURCE_IMAGES_JSON

    val agentPoolId: String
        get() = CloudImageParameters.AGENT_POOL_ID_FIELD

    companion object {
        const val ENVIRONMENT = "environment"
        const val TENANT_ID = "tenantId"
        const val CLIENT_ID = "clientId"
        const val CLIENT_SECRET = "clientSecret"
        const val SUBSCRIPTION_ID = "subscriptionId"
        const val DEPLOY_TARGET = "deployTarget"
        const val REGION = "region"
        const val GROUP_ID = "groupId"
        const val IMAGE_TYPE = "imageType"
        const val IMAGE_URL = "imageUrl"
        const val IMAGE_ID = "imageId"
        const val INSTANCE_ID = "instanceId"
        const val OS_TYPE = "osType"
        const val NETWORK_ID = "networkId"
        const val SUBNET_ID = "subnetId"
        const val MAX_INSTANCES_COUNT = "maxInstances"
        const val VM_SIZE = "vmSize"
        const val VM_NAME_PREFIX = "vmNamePrefix"
        const val VM_PUBLIC_IP = "vmPublicIp"
        const val VM_USERNAME = "vmUsername"
        const val VM_PASSWORD = "vmPassword"
        const val REUSE_VM = "reuseVm"
        const val STORAGE_ACCOUNT_TYPE = "storageAccountType"
        const val TEMPLATE = "template"
        const val TAG_SERVER = "teamcity-server"
        const val TAG_PROFILE = "teamcity-profile"
        const val TAG_SOURCE = "teamcity-source"
        const val TAG_DATA_HASH = "teamcity-data-hash"
        const val TAG_IMAGE_HASH = "teamcity-image-hash"
        const val PROFILE_ID = "profileId"
    }
}
